src/Main.hs

```diff
@@ -71,7 +71,7 @@ backend = Backend'
   , preCompile            = checkConfig
   , postCompile           = \ _ _ _ -> return ()
   , preModule             = moduleSetup
-  , postModule            = writeModule
+  , postModule            = \opts env isM m out -> verifyOutput opts env isM m out >> writeModule opts env isM m out
```

Comment:
Maybe you could just make `verifyOutput` a part of `writeModule`, or else add a new function that does these two functions in sequence?

Reply:
I think a new function is probably best. Having `writeModule`, in `Render`, also do final verification feels like a weird mixup of concerns